pkg/kv/kvserver/rangefeed/stream.go

```diff
@@ -61,22 +56,22 @@ func (s *PerRangeEventSink) SendUnbuffered(event *kvpb.RangeFeedEvent) error {
 		RangeID:        s.rangeID,
 		StreamID:       s.streamID,
 	}
-	return s.wrapped.SendUnbuffered(response)
+	return s.wrapped.send(response, nil)
 }
 
-// Disconnect implements the Stream interface. It requests the UnbufferedSender
-// to detach the stream. The UnbufferedSender is then responsible for handling
-// the actual disconnection and additional cleanup. Note that Caller should not
-// rely on immediate disconnection as cleanup takes place async.
-func (s *PerRangeEventSink) Disconnect(err *kvpb.Error) {
+// Disconnect implements the Stream interface.
+func (s *PerRangeEventSink) SendError(err *kvpb.Error) {
 	ev := &kvpb.MuxRangeFeedEvent{
 		RangeID:  s.rangeID,
 		StreamID: s.streamID,
 	}
 	ev.MustSetValue(&kvpb.RangeFeedError{
 		Error: *transformRangefeedErrToClientError(err),
 	})
-	s.wrapped.SendBufferedError(ev)
+	if err := s.wrapped.send(ev, nil); err != nil {
+		log.Errorf(context.Background(),
```

Comment:
It'll be annoying, but I think we should look into threading a context through some of our send methods as a follow up. We seem to both want to log and need a context for releasing allocations. It would be much nicer for these log messages to have proper log tags which they won't have with context.Background().

Reply:
Good idea, I will file an issue as a follow up on this one.